seesaw/indices/multiscale/preprocessor.py

```diff
@@ -245,13 +246,22 @@ def preprocess_dataset(
             .remote(jit_path=model_link, output_dir=vector_path, meta_dict=meta_dict)
             for i in range(nactors)
         ]
+        print("actors set")
+        file_meta = FastFileMetadataProvider()
+        print("made file meta")
+        rds = ray.data.read_binary_files(
+            paths=read_paths, include_paths=True, parallelism=400, meta_provider=file_meta
+        ).split(200)
 
+        '''
         rds = ray.data.read_binary_files(
-            paths=read_paths, include_paths=True, parallelism=400
+            paths=read_paths, include_paths=True, parallelism=400, meta_provider=file_meta
         ).split(nactors, locality_hints=actors)
-
+        '''
+        # Split into 100, make a repetitive list of 100 actors by multiplying the list
+        rep_actors = actors * int(200/nactors)
```

Comment:
you want some `assert len(rep_actors) >= len(rds)` 
and, you need to make sure  len(rep_actors) >= len(rds), because if it is <, then zip() will stop when rep_actors is done, and the last few partitions of rds will never be processed.
 
I think one way to do this is to do actors * int(math.ceil(200/nactors))